nautobot/users/views.py

```diff
@@ -213,6 +226,130 @@ def post(self, request):
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        The detail view for a saved view should the related ObjectListView with saved configurations applied
+        """
+        instance = self.get_object()
+        query_string = instance.view_config
+        list_view_url = reverse(instance.view) + query_string + f"&saved_view={instance.pk}"
+        return redirect(list_view_url)
+
+    def update(self, request, *args, **kwargs):
+        """
+        This method will extract filter_params, pagination and sort_order from request.GET and apply it to the SavedView specified
+        """
+        sv = get_object_or_404(SavedView, pk=request.GET.get("saved_view", None))
+        table_changes_pending = request.GET.get("table_changes_pending", False)
+        pagination_count = request.GET.get("per_page", None)
+        if pagination_count is not None:
+            sv.config["pagination_count"] = int(pagination_count)
+        sort_order = request.GET.getlist("sort", [])
+        sv.config["sort_order"] = sort_order
+
+        filter_params = {}
+        for key in request.GET:
+            if key in self.non_filter_params:
+                continue
+            filter_params[key] = request.GET.getlist(key)
+        sv.config["filter_params"] = filter_params
+
+        if table_changes_pending:
+            table_class = self.get_table_class_string_from_view_name(sv.view)
+            if table_class:
+                if sv.config.get("table_config", None):
+                    sv.config["table_config"] = {}
+                sv.config["table_config"][f"{table_class}"] = request.user.get_config(f"tables.{table_class}")
+
+        sv.validated_save()
+        list_view_url = sv.get_absolute_url()
+        messages.success(request, f"Successfully updated current view {sv.name}")
+        return redirect(list_view_url)
+
+    def create(self, request, *args, **kwargs):
+        """
+        This method will extract filter_params, pagination and sort_order from request.GET
+        and the name of the new SavedView from request.POST to create a new SavedView.
+        """
+        name = request.POST.get("name")
+        view_name = request.GET.get("view")
+        derived_view_pk = request.GET.get("saved_view", None)
+        derived_instance = None
+        if derived_view_pk:
+            derived_instance = self.get_queryset().get(pk=derived_view_pk)
+        table_changes_pending = request.GET.get("table_changes_pending", False)
+        try:
+            sv = SavedView.objects.create(name=name, owner=request.user, view=view_name)
+        except IntegrityError:
+            messages.error(request, f"You already have a Saved View named '{name}' for this view '{view_name}'")
+            if derived_view_pk:
+                return redirect(self.get_return_url(request, obj=derived_instance))
+            else:
+                return redirect(reverse(view_name))
+        pagination_count = request.GET.get("per_page", 50)
```

Comment:
Presumably this should default to the user's saved default pagination count, not a hard-coded 50?